### PR TITLE
Rooms: Merge privacy setting code into setPrivate

### PR DESCRIPTION
**Pull request by mia-pi-git**
As per https://github.com/smogon/pokemon-showdown/pull/7687#issuecomment-725795911.

### CODE REVIEW

File: server/rooms.ts [hunk 1]
```diff
@@ -703,6 +703,27 @@ export abstract class BasicRoom {
...
+			for (const user of Object.values(this.users)) {
+				if (!user.named) {
+					user.leaveRoom(this.roomid);
+					user.popup(`The battle <<${this.roomid}>> has been made private; you must log in to watch private battles.`);
+				}
+			}
+			if (this.roomid.endsWith('pw')) return true;
+			this.rename(this.title, `${this.roomid}-${password}pw` as RoomID, true);
+		} else if (!privacy && 'makePublic' in this) {
+			(this as GameRoom).makePublic();
```
Comment (Zarel) [inline server/rooms.ts:724]: Why didn't you merge `makePublic` into this?

File: server/chat-commands/room-settings.ts [hunk 2]
```diff
@@ -668,10 +668,8 @@ export const commands: ChatCommands = {
 		const targetRoom = Rooms.search(target);
 		if (!targetRoom) throw new Error(`Error in room creation.`);
 		if (cmd === 'makeprivatechatroom') {
-			targetRoom.settings.isPrivate = true;
+			targetRoom.setPrivate(true);
 			if (!targetRoom.persist) throw new Error(`Private chat room created without settings.`);
```
Comment (Zarel) [inline server/chat-commands/room-settings.ts:672]:
```suggestion
			if (!targetRoom.persist) throw new Error(`Private chat room created without settings.`);
			targetRoom.setPrivate(true);
```

File: server/chat-commands/room-settings.ts [hunk 3]
```diff
@@ -668,10 +668,8 @@ export const commands: ChatCommands = {
 		const targetRoom = Rooms.search(target);
 		if (!targetRoom) throw new Error(`Error in room creation.`);
 		if (cmd === 'makeprivatechatroom') {
-			targetRoom.settings.isPrivate = true;
 			if (!targetRoom.persist) throw new Error(`Private chat room created without settings.`);
```
Comment (Zarel) [inline server/chat-commands/room-settings.ts:671]: Does this even need to be thrown? This just using `/makeprivatechatroom` on a groupchat/battle, right?

Comment (mia-pi-git) [reply server/chat-commands/room-settings.ts:671]: Looks like it. This shouldn't be necessary from what it seems, i agree.

Comment (Zarel) [reply server/chat-commands/room-settings.ts:671]: After some testing, this doesn't look like a problem, but it's definitely pretty confusing code.

**Pull request merged**